### ISSUE TITLE
Switch toString to LexicalPreservingPrinter when configured

### ISSUE THREAD

**Issue by matozoid** [javaparser/javaparser#1821]

**Comment by jhalliday** [comment]
On the basis of the API design principle of 'least surprise', I expect that

given
parserConfiguration.setLexicalPreservationEnabled(true);
when
String outputString = JavaParser.parse(inputString).toString();
then
inputString.equals(outputString) == true

However that doesn't work, because cu is not aware of the parser config. Round-trip instead requires code of the form:

outputString = LexicalPreservingPrinter.print(cu);

which, although workable, seems somewhat inelegant.

**Comment by matozoid** [comment]
toString has evolved from a quick dump of the AST as Java code to the pretty printer it is calling now. I don't think there's another reason for having the pretty printer there than "it has always been like that."

I think the Lex Pres Printer already stores data in the AST, so it wouldn't even be that difficult to detect when to use it.

**Comment by matozoid** [comment]
Since this is a structural discussion, I'd like to hear @ftomassetti 's opinion on this.

**Comment by ftomassetti** [comment]
I agree that it should use LexicalPreservation. I think it is what most people want anyway. Perhaps I would add a way to force pretty printing like a `toPrettyPrintedString()` in case people want to use it even if they have lexical preservation enabled.

**Comment by matozoid** [comment]
There's a PrettyPrinter class meant to do that, so that's a solved problem :-)

Let's follow @jhalliday's suggestion then.

**Comment by ThLeu** [comment]
I would agree that it should become the standard for `toString()` and that another method like `toPrettyPrintedString()` should be there aswell for convenience at least.
Of course this is a breaking change which has to be considered but in the long run it will make more people use LexicalPreservation hopefully :).
